tt_metal/llrt/tlb_config.cpp

```diff
@@ -93,24 +99,36 @@ int32_t get_static_tlb_index(CoreCoord target) {
     // implementation migrated from blackhole.py in `src/t6ifc/t6py/packages/tenstorrent/chip/blackhole.py` from tensix
     // repo (t6py-blackhole-bringup branch)
 
-    // BH worker cores are starting from y=2
-    target.y--;
-    if (is_eth_location) {
-        // TODO: Uplift this
-        return -1;
-    } else if (is_tensix_location) {
-        if (target.x >= 8) {
-            target.x -= 2;
-        }
+    auto dram_tlb_index =
+        std::find(tt::umd::blackhole::DRAM_LOCATIONS.begin(), tt::umd::blackhole::DRAM_LOCATIONS.end(), target);
+    if (dram_tlb_index != tt::umd::blackhole::DRAM_LOCATIONS.end()) {
+        auto dram_index = dram_tlb_index - tt::umd::blackhole::DRAM_LOCATIONS.begin();
+        // We have 3 ports per DRAM channel so we divide index by 3 to map all the channels of the same core to the same TLB
+        return tt::umd::blackhole::TLB_BASE_INDEX_4G + (dram_index / NUM_PORTS_PER_DRAM_CHANNEL);
```

Comment:
Just to confirm, this logic means that static TLBs mapped to DRAM are all 4G in size, tensix TLBs start at idx 38 and eth start at idx 0?

Just curious: do we have a file/spreadsheet showing all the TLB assignments? It usually makes debug a lot easier.

Reply:
yep, the start indices was pulled from a syseng script for tlb assignment. I have seen the tlb spreadsheets for gs/wh but good point, Ill try to locate the one for BH